Bio/SeqIO/XdnaIO.py

```diff
@@ -156,9 +163,13 @@ def XdnaIterator(handle):
     # as I know, so we ignore that value. SerialCloner has no such concept
     # either and always generates files with a neg_length of zero.
 
-    with as_handle(handle, "rb") as handle:
+    try:
 
-        header = _read_header(handle, 112)
+        header = handle.read(112)
+        if not header:
+            raise ValueError("Empty file.")
+        if len(header) < 112:
+            raise ValueError("Improper header, cannot read 112 bytes from handle" % length)
```

Comment:
```
$ flake8
./Bio/SeqIO/XdnaIO.py:172:30: BLK100 Black would make changes.
./Bio/SeqIO/XdnaIO.py:172:85: F821 undefined name 'length'
```

This line is currently 91 characters long, so black would break that - but we don't need the ``% length`` bit which also solves the long line.